packages/cache/src/internal/cacheUtils.ts

```diff
@@ -82,19 +83,27 @@ async function getVersion(app: string): Promise<string> {
 
 // Use zstandard if possible to maximize cache performance
 export async function getCompressionMethod(): Promise<CompressionMethod> {
-  const versionOutput = await getVersion('zstd')
-  return versionOutput.toLowerCase().includes('zstd command line interface')
-    ? CompressionMethod.Zstd
-    : CompressionMethod.Gzip
+  if (process.platform === 'win32' && !isGnuTarInstalled()) {
+    // Disable zstd due to bug https://github.com/actions/cache/issues/301
+    return CompressionMethod.Gzip
+  } else {
+    const versionOutput = await getVersion('zstd')
+    const version = semver.clean(versionOutput)
+    return !versionOutput.toLowerCase().includes('zstd command line interface')
+      ? CompressionMethod.Gzip
+      : !version || semver.lt(version, 'v1.3.2')
+      ? CompressionMethod.ZstdOld
+      : CompressionMethod.Zstd
```

Comment:
This block of quote is a little difficult to parse. It might be better to break this up and add some comments about why we care about `v1.3.2`

```suggestion
    if (!versionOutput.toLowerCase().includes('zstd command line interface')) {
      return CompressionMethod.Gzip
    }
    
    // v1.3.2 is required to use the `--long` options in ZSTD
    if (!version || semver.lt(version, 'v1.3.2') {
      return CompressionMethod.ZstdOld
    }
    
    return CompressionMethod.Zstd
```